### DISCUSSION TITLE
How to set priority of variables in CP-SAT solver

### DISCUSSION THREAD

**Discussion by pardeep632** [google/or-tools#3318] [CP-SAT questions] [answered]
**What version of OR-Tools and what language are you using?**
Version: 9.3.10497.
Language: JAVA

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
CP-SAT

**What operating system (Linux, Windows, ...) and version?**
Mac OS Monterey

How can i set the priority of variables in CP-SAT. I have a variable Array say , x1+x2+x3+x4...x11 etc i want to maximize it ..Each variable can be either 0 or 1. I also have a constraint like ..
model.addLessOrEqual(LinearExpr.weightedSum(variableArray, dataList), maxValue)

i want to set some priority on each variable, which should decide if the variable is 1 or 0 in the final solution

**Comment by lperron** [comment]
Why don't you trust the search procedure ? Especially with multiple workers.

**Reply by pardeep632** [reply]
I am not getting the expected result using search procedure. I will explain the requirement. I have few variables which take up 0 or 1 value. each variable has some data associated with it . I want that weighted some of variables with this data list should not exceed some maxium value. As represented below:
model.addLessOrEqual(LinearExpr.weightedSum(variableArray, dataList), maxValue)

Now there is a second data List associated with the variables , I want that the variables which are set to 1 after search procedure , should give me the sorted list of second data list. For e.g: If 
X1, X2, X3,X4 are variables
200, 400, 200, 100 is data list
maximum value is 650
20, 13, 14, 15 is the second data list 
i want that x1, x4, X3 should be set to one, because it gives me : 20,15,14 as second list and sum as 200+100+200

May be my objective is not correct , but i am not sure how to achieve this

**Comment by lperron** [comment, marked as answer]
search will not guarantee any properties. Desired outcome should be in the constraints or in the objective.
